lib/cdo/aws/cloud_formation.rb

```diff
@@ -220,6 +222,114 @@ def create_or_update
         end
       end
 
+      def start_inactive_instance
+        cloudformation_resource = Aws::CloudFormation::Resource.new
+        stack = cloudformation_resource.stack(stack_name)
+        instance = Aws::EC2::Instance.new(id: stack.resource('WebServer').physical_resource_id)
+        if instance.state.code != 80
+          CDO.log.info "Instance #{instance.id} in Stack #{stack_name} can't be started because it is not" \
+          " currently stopped.  Current state - #{instance.state.code}:#{instance.state.name}"
+        else
+          CDO.log.info "Starting Instance #{instance.id} ..."
+          instance.start
+          instance.wait_until_running
+          CDO.log.info "Instance #{instance.id} is started."
+
+          public_ip_address = instance.reload.public_ip_address
+          dashboard_url = stack.outputs.select {|output| output.output_key == 'DashboardURL'}.first.output_value
+          pegasus_url = stack.outputs.select {|output| output.output_key == 'PegasusURL'}.first.output_value
+
+          # suffix period to construct fully qualified domain name
+          pegasus_domain_name = URI.parse(pegasus_url).host + '.'
+          dashboard_domain_name = URI.parse(dashboard_url).host + '.'
+
+          route53_client = Aws::Route53::Client.new
+
+          # this lookup may stop working if/when there are more than 100 zones
+          # prefix zone name with a period to prevent partial match (don't let zone "code.org." match "foo.cdn-code.org.")
+          hosted_zone_id = route53_client.
+            list_hosted_zones.
+            hosted_zones.
+            select {|zone| pegasus_domain_name.end_with?('.' + zone.name)}.
+            first.
+            id
```

Comment:
Instead of iterating through all of the hosted zones, you might be able to use [`list_hosted_zones_by_name`](https://docs.aws.amazon.com/sdkforruby/api/Aws/Route53/Client.html#list_hosted_zones_by_name-instance_method), like:
```ruby
zone_name = URI.parse(pegasus_url).host.split('.').last(2).join('.') + '.'
hosted_zone_id = route53_client.
  list_hosted_zones_by_name(
    dns_name: zone_name,
    max_items: 1
  ).
  hosted_zones.
  first.
  id
```

Reply:
I was avoiding this type of lookup in the unlikely event that a 3rd level domain (`hourofcode.org.uk.`) was being used as the zone for the stack.

Reply:
ah, good point! looks like there's no general-purpose way to extract the domain from a hostname without querying central name-servers, so the current approach sounds good.